src/thunks/cleanTrashedGroups.js

```diff
@@ -1,8 +1,11 @@
+import { Q } from 'cozy-client'
 import { DOCTYPE_CONTACT_GROUPS, DOCTYPE_CONTACTS } from '../helpers/doctypes'
 
 const cleanTrashedGroups = () => async (dispatch, getState, { client }) => {
   const { data: trashedGroups } = await client.query(
-    client.find(DOCTYPE_CONTACT_GROUPS).where({ trashed: true })
+    Q(DOCTYPE_CONTACT_GROUPS)
+      .where({ trashed: true })
```

Comment:
This should be in a partial index: as it is a static predicate, there is no gain to use a selector

Reply:
I agree, a lot of refactoring/optimization work is possible.